bench_manager/bench_manager/doctype/bench_settings/bench_settings.py

```diff
@@ -399,15 +421,278 @@ def backup_sites_with_monthly_option():
+	except JobTimeoutException:
+		if retry_count < 2:
+			args = {
+				"retry_count": retry_count + 1,
+				"upload_db_backup": False,  # considering till worker timeout db backup is uploaded
+			}
+			enqueue(
+				"bench_manager.bench_manager.doctype.bench_settings.bench_settings.take_backup_to_dropbox",
+				queue="long",
+				timeout=1500,
+				**args
+			)
+	except Exception:
+		error_message = frappe.get_traceback()
+		send_email(False, "Dropbox", "Bench Settings", "send_notifications_to", error_message)
+
+
+@frappe.whitelist()
+def get_dropbox_authorize_url():
+	app_details = get_dropbox_settings(redirect_uri=True)
+	dropbox_oauth_flow = dropbox.DropboxOAuth2Flow(
+		consumer_key=app_details["app_key"],
+		redirect_uri=app_details["redirect_uri"],
+		session={},
+		csrf_token_session_key="dropbox-auth-csrf-token",
+		consumer_secret=app_details["app_secret"],
+	)
+
+	auth_url = dropbox_oauth_flow.start()
+
+	return {"auth_url": auth_url, "args": parse_qs(urlparse(auth_url).query)}
+
+
+@frappe.whitelist()
+def get_redirect_url():
+	if not frappe.conf.dropbox_broker_site:
+		frappe.conf.dropbox_broker_site = "https://dropbox.erpnext.com"
+	url = "{0}/api/method/dropbox_erpnext_broker.www.setup_dropbox.get_authotize_url".format(
+		frappe.conf.dropbox_broker_site
+	)
+
+	try:
+		response = make_post_request(url, data={"site": get_url()})
+		if response.get("message"):
+			return response["message"]
+
+	except Exception:
+		frappe.log_error()
+		frappe.throw(
+			_(
+				"Something went wrong while generating dropbox access token. Please check error log for more details."
+			)
+		)
+
+
+def backup_to_dropbox(site_list,upload_db_backup=True):
+	if not frappe.db:
+		frappe.connect()
+
+	# upload database
+	dropbox_settings = get_dropbox_settings()
+
+	if not dropbox_settings["access_token"]:
+		access_token = generate_oauth2_access_token_from_oauth1_token(dropbox_settings)
+
+		if not access_token.get("oauth2_token"):
+			return (
+				"Failed backup upload",
+				"No Access Token exists! Please generate the access token for Dropbox.",
+			)
+
+		dropbox_settings["access_token"] = access_token["oauth2_token"]
+		set_dropbox_access_token(access_token["oauth2_token"])
+
+	dropbox_client = dropbox.Dropbox(
+		oauth2_access_token=dropbox_settings["access_token"], timeout=None
+	)
+
+	if upload_db_backup:
+			if site_list:
+				create_backup(site_list)
+				sync_backups()
+				for i in site_list:
+					last_doc = frappe.get_list("Site Backup", filters={'site_name':i.name}, fields=['file_path'], order_by="creation desc",limit = 1)[0]
+					list_string = last_doc.file_path.split('/')
+					list_string.pop(0)
+					list_string.insert(0, '.')
+					string_list = ("/".join(list_string))
+					upload_file_to_dropbox(string_list+"-database.sql.gz", "/database", dropbox_client)
+					upload_file_to_dropbox(string_list+"-site_config_backup.json", "/database", dropbox_client)
+
+
+def upload_file_to_dropbox(filename, folder, dropbox_client):
```

Comment:
this method should called for uploaded flies to the sile like images and pdfs
where is this call?